### PR TITLE
instrument gql query retry

### PR DESCRIPTION
**Pull request by jmelis**
I want to make sure we capture all errors on every gql attempt. For that we need the hook capability of the `@retry` method available in a newer sretoolbox release. I was force to update semver as sretoolbox depends on a newer version of semver.

### CODE REVIEW

File: reconcile/utils/gql.py [hunk 1]
```diff
@@ -91,7 +92,7 @@ def __init__(self, url, token=None, int_name=None, validate_schemas=False):
             if not self._valid_schemas:
                 raise GqlApiIntegrationNotFound(int_name)
 
-    @retry(exceptions=GqlApiError, max_attempts=5)
+    @retry(exceptions=GqlApiError, max_attempts=5, hook=capture_exception)
```
Comment (rporres) [inline reconcile/utils/gql.py:95]: what happens if `capture_exception` fails?

Comment (rporres) [reply reconcile/utils/gql.py:95]: Or even better, we have to make sure that it is fire and forget.

File: reconcile/utils/gql.py [hunk 2]
```diff
@@ -29,6 +30,19 @@
 """
 
 
+def capture_and_forget(error):
```
Comment (jmelis) [inline reconcile/utils/gql.py:33]: is this what you had in mind @rporres?

**Comment by jmelis** [comment]
FYI I had to bump semver to the version that sretoolbox uses, which generated some warnings that I had to fix by stopping using `semver.format_version` and `semver.parse`

File: reconcile/utils/gql.py (hunk 2 above)
Comment (rporres) [reply reconcile/utils/gql.py:33]: I guess that makes me feel more relaxed about this  :-D

**Review by rporres — APPROVED** [review]

**Pull request merged**